R/contact_matrix.r

```diff
@@ -419,13 +418,31 @@ contact_matrix <- function(survey, countries=c(), survey.pop, age.limits, filter
                         survey$participants[, median(get(columns[["year"]]), na.rm=TRUE)]
                 }
             }
+        } else{
+            # if survey.pop is a data frame with columns 'lower.age.limit' and 'population'
+            survey.pop <- data.table(survey.pop)
+            # make sure the maximum survey.pop age exceeds the participant age group breaks
+            if(max(survey.pop$lower.age.limit) < max(part.age.group.present)){
+                survey.pop <- rbind(survey.pop,
+                                    list(max(part.age.group.present+1),0))
+            }
+            
+            # add dummy survey.year
+            survey.year <- NA
         }
+        
+        # add upper.age.limit after sorting the survey.pop ages (and add maximum age > given ages)
+        survey.pop <- survey.pop[order(lower.age.limit),]
+        survey.pop$upper.age.limit <- unlist(c(survey.pop[-1,'lower.age.limit'],
```

Comment:
another minor point: can we use double quotation marks to be in line with the rest of the code (unless there's a reason not to)?

Reply:
done